models/clm/readme.rst

```diff
@@ -1,40 +1,195 @@
+for some examples. The list is by no means complete. Each experiment has extended
+the capabilities of the CLM-DART system. The observations supported by
+DART are contained in the *obs_def_land_mod.f90*, *obs_def_tower_mod.f90*, 
+and *obs_def_COSMOS_mod.f90* modules - each of which may be easily extended. 
+New observation types are usually easily supported. There are **many** examples 
+of routines to convert from native data formats (netCDF, HDF, csv, etc.) 
+to a DART observation sequence file.
+
+
+Important Features
+------------------
+
+Land DA is extremely diverse. The support for Land DA as pertains to CLM-DART
+has some features that need to be described in some detail.
+
+
+CLM indeterminate values
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+CLM variables are encoded as rectangular arrays in the netCDF files. (Thankfully!)
+However, this means that some variables have space for layers that are unused.
+Anything with snow layers, for example. CLM has the *SNLSNO* varible to indicate
+which snow layers are active. The *unused* layers may not have the *_FillValue*
+value, but can have 'indeteminate' values. The :doc:`clm_to_dart` 
+must be run to convert these indeterminate values to *_FillValue* to be 
+interpreted correctly by DART.  After the assimilation is complete, the 
+:doc:`dart_to_clm` must be called to replace the *_FillValue* with whatever
+is originally in that slot. Extra care is taken to handle the special case
+of a *trace* of snow. See the *Discussion of Indeterminate Values* 
+section of :doc:`clm_to_dart` for more details.
+
+
+Model Interpolate - The Forward Operator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since the subgridscale components of CLM have no explicit location associated 
+with them, the location of every component in the gridcell is the same as the 
+gridcell itself. The DART forward operators fundamentally rely on 
+interpolating the model state to some arbitrary location. At present, the best we
+can do is to create an area-weighted average of all pieces in the gridcell.
+This is clearly sub-optimal. A nice project would be to use some sort of lookup
+table for the observation location to determine the dominant PFT or whatever at
+that location and then just average up all the similar PFTs in the gridcell.
+The forward operator would be more accurate and might have a discernable impact
+on the regression relationship (i.e. ensemble covariance) with the variables
+in the DART state vector.
+
+The *model_interpolate* function in DART achieves efficiency by interpolating
+all the ensemble members at the same time. This gives rise to some challenging
+problems when interpolating values for variables with dynamic active layers.
+Some ensemble members may only have 2 active snow layers, some may have 3 ...
+this is an untenable situation when asked for the snow temperature or water 
+content in layer 3, for example. Consequently - *model_interpolate* will fail
```

Comment:
... but for water content, for example, the situation can become more tenable if we interpolate the column total water content, instead of the individual layer content, correct?  Because interpolating the total water content does not necessitate the existence of all layers across all ensembles -- just the existence of at least 1 layer.  Then if we regress upon the column total water storage, we don't have to have knowledge of the individual layers, but we do have to create a repartitioning algorithm.... we come to this later in documentation.